shopfloor_reception: fix picking state after auto-posting

jbaudoux opened the pull request:
Picking was in waiting state with the remaining moves to receive instead of assigned state (available)

simahawk commented on shopfloor/models/stock_move.py:
```diff
@@ -39,8 +39,8 @@ def split_other_move_lines(self, move_lines, intersection=False):
             split_move = self.create(split_move_vals)
-            split_move._action_confirm(merge=False)
             split_move.move_line_ids = to_move
+            split_move._action_confirm(merge=False)
```
this is not a a fix in `shopfloor_reception`.  Can you split the commit pls?
Also, wondering: the change seems to make sense but... we didn't have any issue before?

jbaudoux replied:
This is because the supplier location bypass the reservation and `action_assign` is called here https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1177
which causes also the creation of a new move line here
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1324

jbaudoux replied:
This issue is happening only in case of a reception

jbaudoux replied:
I have done the split in 2 commits

jbaudoux committed:
shopfloor: fix split move lines

The move line must be linked to the new split move before calling
_action_confirm.

In case of a reception picking where the supplier location bypass the
reservation, action_assign is called by _action_config here
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1177
which causes the creation of an unwanted new move line here
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L1324
This is why we must relink the existing move line to the new move before
calling _action_confirm.

The new move line was also causing the original picking to change state
to waiting.

jbaudoux committed:
shopfloor_reception: Solved TODO

The underlying issue has been fixed. Removing the temporary workaround.

jbaudoux commented:
@simahawk Can you update your review and merge it if now fine ?

simahawk commented:
@jbaudoux pls rebase

simahawk commented:
/ocabot merge patch
